ffi/athcon/bindings/go/athcon.go

```diff
@@ -188,21 +164,56 @@ type Result struct {
 	GasRefund int64
 }
 
-func (vm *VM) Execute(ctx HostContext, rev Revision,
-	kind CallKind, depth int, gas int64,
-	recipient Address, sender Address, input []byte, value Bytes32,
-	code []byte) (res Result, err error) {
-
-	ctxId := addHostContext(ctx)
-	// FIXME: Clarify passing by pointer vs passing by value.
-	athconRecipient := athconAddress(recipient)
-	athconSender := athconAddress(sender)
-	athconValue := athconBytes32(value)
-	result := C.execute_wrapper(vm.handle, C.uintptr_t(ctxId), uint32(rev),
-		C.enum_athcon_call_kind(kind), C.int32_t(depth), C.int64_t(gas),
-		&athconRecipient, &athconSender, bytesPtr(input), C.size_t(len(input)), &athconValue,
-		bytesPtr(code), C.size_t(len(code)))
-	removeHostContext(ctxId)
+func (vm *VM) Execute(
+	ctx HostContext,
+	rev Revision,
+	kind CallKind,
+	depth int,
+	gas int64,
+	recipient, sender Address,
+	input []byte,
+	value Bytes32,
+	code []byte,
+) (res Result, err error) {
+	if len(code) == 0 {
+		return res, fmt.Errorf("code is empty")
+	}
+	msg := C.struct_athcon_message{
```

Comment:
Let's add a comment here that explains why the message doesn't contain the code, despite having a code field. (Need to double check but I think the `code` field on the message is specifically for deploying new code.)

Reply:
I also was puzzled by this, but ATM I'm not 100% sure when `code` is supposed to be populated. When deploying new code, shouldn't it be placed in `input`?

Reply:
well, we haven't built the deploy opcode yet (#76), we can finalize it when we do that. this design was copied from EVM.